lstchain/io/dataselection.py

```diff
@@ -0,0 +1,263 @@
+    DL3 data reduction
+
+    Parameters for event filters will be combined in a dict so that the
+    filter_events() can be used.
+    """
+
+    intensity = List(
+        help="Range of intensity of event filter",
+        trait=Float(),
+        default_value=[0, np.inf],
+    ).tag(config=True)
+
+    length = List(
+        help="Range of length for event filter",
+        trait=Float(),
+        default_value=[0, np.inf],
+    ).tag(config=True)
+
+    width = List(
+        help="Range of width for event filter",
+        trait=Float(),
+        default_value=[0, np.inf],
+    ).tag(config=True)
+
+    r = List(
+        help="Range of r for event filter",
+        trait=Float(),
+        default_value=[0, 1],
+    ).tag(config=True)
+
+    wl = List(
+        help="Range of wl for event filter",
+        trait=Float(),
+        default_value=[0.01, 1],
+    ).tag(config=True)
+
+    leakage_intensity_width_2 = List(
+        help="Range for leakage_intensity_width_2 of event filter",
+        trait=Float(),
+        default_value=[0, 1],
+    ).tag(config=True)
+
+    fixed_gh_cut = Float(
+        help="Fixed selection cut for gh_score (gammaness)",
+        default_value=0.5,
+    ).tag(config=True)
+
+    fixed_theta_cut = Float(
+        help="Fixed selection cut for theta",
+        default_value=0.5,
+    ).tag(config=True)
+
+    fixed_source_fov_offset_cut = Float(
+        help="Fixed selection cut for source FoV offset",
+        default_value=3,
+    ).tag(config=True)
+
+    src_dep_alpha = Float(
+        help="Selection cut for source dependent parameter - alpha",
+        default_value=8.0,
+    ).tag(config=True)
+
+    lst_tel_ids = List(
+        help="List of selected LST telescope ids",
+        trait=Int(),
+        default_value=[1],
+    ).tag(config=True)
+
+    magic_tel_ids = List(
+        help="List of selected MAGIC telescope ids",
+        trait=Int(),
+        default_value=[1, 2],
+    ).tag(config=True)
+
+    aliases = {
+        ("int", "intensity"): "DataSelection.intensity",
+        ("len", "length"): "DataSelection.length",
+        ("w", "width"): "DataSelection.width",
+        "r": "DataSelection.r",
+        "wl": "DataSelection.wl",
+        (
+            "leak_2",
+            "leakage_intensity_width_2",
+        ): "DataSelection.leakage_intensity_width_2",
+        ("gh", "fixed_gh_cut"): "DataSelection.fixed_gh_cut",
+        ("theta", "fixed_theta_cut"): "DataSelection.fixed_theta_cut",
+        (
+            "src_fov",
+            "fixed_source_fov_offset_cut",
+        ): "DataSelection.fixed_source_fov_offset_cut",
+        ("alpha", "src_dep_alpha"): "DataSelection.src_dep_alpha",
+        "lst_tel_ids": "DataSelection.lst_tel_ids",
+        "magic_tel_ids": "DataSelection.magic_tel_ids",
+    }
+
+    def __init__(self, parent=None, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+
+    def __call__(self):
```

Comment:
Why implement `__call__` as doing nothing?

Reply:
I am not very clear about it. What should it return for this `Component`?

Reply:
Implementing `__call__`  means you can use an instance of this class like a function.

```
data_selection = DataSelection()
data_selection()
```

Since you wrote the class and not me, you should know why it is needed and what it should do, right?

Reply:
For what do you use it? What is it supposed to do?

Reply:
I assumed I can use it for testing, but it is unclear to me how I can or even should, test these `Components`.

Reply:
I have made the call function to do a simple check, whether the data table passed is QTable or not.